File: 5_kyu/RGB_To_Hex_Conversion.c
#include <stdio.h>

// Clip x to the range 0..0xff
static int clip(const int x)
{
  if( x < 0 )
    return 0;
  if( x > 0xff )
    return 0xff;
  return x;
}

// Copies the RGB color in hex into the output buffer.
// Buffer size must be at least 7 chars.
int rgb(const int r, const int g, const int b, char output[])
{
  snprintf( output, 7, "%02X%02X%02X", clip(r), clip(g), clip(b) );
  return 0; 
}

File: 5_kyu/RGB_To_Hex_Conversion.kt
/** Cuts integer value to range 0..255 of a byte */
fun cutToByte(x: Int) =
    when {
        x > 0xff -> 0xff
        x < 0 -> 0
        else -> x
    }

/** Formats rgb channel values into hex color string */
fun rgb(r: Int, g: Int, b: Int) = "%02X%02X%02X".format(cutToByte(r), cutToByte(g), cutToByte(b))
